iop/dev9/atad/src/ps2atad.c

```diff
@@ -197,11 +199,14 @@ static void ata_multiword_dma_mode(int mode);
 static void ata_ultra_dma_mode(int mode);
 static void ata_shutdown_cb(void);
 
+int ata_device_sector_io_internal(int device, void *buf, u64 lba, u32 nsectors, int dir);
+
 #ifdef ATA_ENABLE_BDM
-static int ata_bd_read(struct block_device *bd, u32 sector, void *buffer, u16 count);
-static int ata_bd_write(struct block_device *bd, u32 sector, const void *buffer, u16 count);
+static int ata_bd_read(struct block_device *bd, u64 sector, void *buffer, u16 count);
+static int ata_bd_write(struct block_device *bd, u64 sector, const void *buffer, u16 count);
```

Comment:
even if we have to change the prototype for these functions, couldn't we take the opportunity to standardize them to the POSIX.1-2001 standard by using the following prototypes: 
ssize_t pread(int fd, void *buf, size_t count, off_t offset);
ssize_t pwrite(int fd, const void *buf, size_t count, off_t offset);

Reply:
No, pread and pwrite can write byte sized data and we do not (or want to) support that. Also, our read/write functions are part of the object oriented block_device structure, so they need the block_device structure as their first argument.